### ISSUE TITLE
Add support for 2nd Slot in ClickPLC Plus

### ISSUE THREAD

**Issue by ssweber** [alexrudd2/clickplc#47]
@alexrudd2 this is something I'll investigate. It's not something I actively need. Just poking around :)

Additional logic needs to be added to `_get_x`, `_get_y` and possibly `_set_y` to special case it.

Combo Input/Output are 8 in/6 out.
Combo Input/Output + Analog, are 4/4 + df spaces.

![image](https://github.com/user-attachments/assets/08bbdedb-8110-4a59-9f6f-4b06fa3b913c)

**Comment by alexrudd2** [comment]
If the MIT license is for inter-op, you can always dual-license (A)GPL + MPL.  :)   Having a formal GPL license would make my life a bit easier for merging, well if anyone still cares about copyright enforcement in the age of LLMs, anyways....

Huh, I never had contact with the Click engineer.  I think I was talking to a PM.   Interesting!  I'm kinda curious what secrets that old codebase holds.

If you want to explore ladder logic and code, etc, have you looked at CODESYS?  It doesn't get much usage in the states, but is a much more open platform over in Europe, and I think there's a simulator built-in.  https://en.wikipedia.org/wiki/CODESYS.  I think there are even variants that run on RPi and maaaaybe even that ProductivityOpen Arduino.  If nothing else, go read IEC 61131 to avoid re-inventing the wheel on ladder logic. :)

Haven't really looked at ClickNic yet, so I'm not quite sure what you're reverse engineering.  But for trying to reverse engineer some Ethernet.IP stuff I found ImHex (https://imhex.werwolv.net/) to be amazing.  LLMs understand its little syntax decently, as well!

**Comment by alexrudd2** [comment]
Also, you can get my email (it ends with `.tech`) from the commits in this repo.

**Comment by ssweber** [comment]
@alexrudd2 I sent you an email. Anyway... I went way down the rabbit hole and my [laddercodec](https://github.com/ssweber/laddercodec) library is my reverse-engineered decoding/encoding of CLICK PLC editor's clipboard/scr*.tmp files. It's getting pretty stable, and considering you work at Copia, thought it might interest you. 

I've built the functionality into ClickNick (the Click editor companion gui), so you can export the open Click Project to csv that looks like:

<img width="676" height="205" alt="Image" src="https://github.com/user-attachments/assets/1285de27-7af7-4c2c-9aa8-e4676b2f92fd" />

translates to this
```
marker,A,B,C,D,E,F,G,H,I,J,K,L,M,N,O,P,Q,R,S,T,U,V,W,X,Y,Z,AA,AB,AC,AD,AE,AF
R,C1,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,"on_delay(T1,TD1,preset=1,unit=Ts)"
```

and all the way back in:

<img width="487" height="328" alt="Image" src="https://github.com/user-attachments/assets/ddbc83ed-456b-4cbe-82c7-802b2cb4f54d" />

Diff, git, etc. (it's just plain text)

then (if you really want to), convert it to pyrung via codegen (the codegen will attach your nickname/tag names if so provided):

```python
with Rung(Filling):
    on_delay(FillTmr, preset=10, unit="Ts").reset(C2)
```

and use pytest to make assertions, reason through code, etc. Run it, convert it back to csv, import it back in. You get the idea.

**Comment by ssweber** [comment]
More complicated output looks like this (the `#,` rows are Rung comments that get put into the csv too:)

```
#,  7. OR + two branches
#,
#,"  with Rung(Or(X001, X002)):"
#,      out(Y001)
#,      with branch(C1):
#,          out(Y002)
#,      with branch(C2):
#,          out(Y003)
R,X001,T,T,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,out(Y001)
,X002,,T:C1,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,out(Y002)
,,,C2,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,-,out(Y003)
```

<img width="673" height="330" alt="Image" src="https://github.com/user-attachments/assets/4ca8dabc-b825-4b74-8702-0b538b70f2ca" />

I'll stop spamming your project's issue board now 😆 .

**Comment by alexrudd2** [comment]
Yeah, I've been meaning to reply to that email but it's a lot to take in at once, sorry 😄. I hope to get some breathing room soon
